.jenkins/checksum.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo $(md5sum $@ | md5sum | awk '{print $1;}')
```

Comment:
what was the reasoning to exposing this on a higher level as compared to hiding it in `cache.sh`? I like that we have very fine grained control over what exactly is required to build the hash. Do we not consider this an implementation detail though?

Reply:
The short answer is because this is how CircleCI designed it.

The longer answer is, the files we are hashing as the cache key are entirely different than the files being cached, so we would need to add those as arguments to cache.sh. This isolates the tasks of defining a cache key and actually doing the caching into two scripts. The cache key can be anything - it doesn't have to be a file hash. Here we also add things like `v1` and whether this is for GPU or CPU.

Reply:
that makes sense